cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1859,14 +1861,23 @@ cabalPkgname = PackageName "Cabal"
 basePkgname  = PackageName "base"
 
 
-legacyCustomSetupPkgs :: Platform -> [PackageName]
-legacyCustomSetupPkgs (Platform _ os) =
+legacyCustomSetupPkgs :: Compiler -> Platform -> [PackageName]
+legacyCustomSetupPkgs compiler (Platform _ os) =
     map PackageName $
         [ "array", "base", "binary", "bytestring", "containers"
-        , "deepseq", "directory", "filepath", "pretty"
-        , "process", "time" ]
+        , "deepseq", "directory", "filepath", "old-time", "pretty"
+        , "process", "time", "transformers" ]
+     --TODO: [research required] does old-time need to be conditional?
      ++ [ "Win32" | os == Windows ]
      ++ [ "unix"  | os /= Windows ]
+     ++ [ "ghc-prim"         | isGHC ]
+     ++ [ "template-haskell" | isGHC ]
+  where
+    -- Have to be slightly clever with the compatibility checks to take into
+    -- account compilers that claim compatibility with each other, like GHCJS.
+    isGHC = not $ null [ () | CompilerId GHC _ <- compilerIds ]
```

Comment:
Added a new util. Turns out we had a similar one already, so added a variant in that style.